labs/yui.md

```diff
@@ -0,0 +1,78 @@
+
+This allows cross-chain message relaying not only to the trusted and privileged actors, but also to any actor that satisfies the ACL of the permissioned chain.
+
+In addition, we aim to apply the method not only to the permissioned chains, but to the public chains as well.
+
+### Provide a unified communication method independent from a specific blockchain/DLT implementation.
+This enables application developers to develop chain agnostic protocols.
+
+### Support arbitrary data transfer and computation
+Developer can design and implement the specific protocol for its cross-chain application, not limited to token transfer use case.
+
+### Avoid adding components that introduce additional trust (apart from individual blockchains)
+Introducing such components may decrease the level of security of a system as it would be bounded by the lowest component in the system.
+
+Also, introducing new (off-chain) components with high fault tolerance will also lead to increased operational costs
+
+### Do not introduce a new layer
+Each actor (of ledgers) has only to interact with a respective ledger of its interest to complete a cross-chain operation unless the application has additional requirements.
+
+## Use cases
+Cross-chain smart contracts for arbitrary data and procedures, not limited to atomic swap of tokens.
+- Cross-chain token transfer
+- Cross-chain Atomic swap
+  - Example includes delivery versus payment settlement
+- Cross-chain atomic execution of arbitrary procedures
+
+## Current status and future plan
+We have already developed modules for Hyperledger Fabric, Hyperledger Besu, and Corda. We plan to apply these modules for practical applications as well as to expand support for other blockchains.
+- At first, we are focusing on permissioned blockchain environment
+
+Aim to enable developers in Hyperledger community to introduce interoperability in their systems and develop cross-chain applications without too many changes and increase the complexity
+
+## Components
+Client (on-chain) module and Relayer to support various ledgers in align with the design principles
+- Client module should be designed to support each underlying ledger
+
+Various software to support the development of cross-chain systems
+- Modules to support the development of cross-chain contracts
+  - Implementation of an atomic commit protocol for cross-chain atomic operation
+  - Implementation of authentication for cross-chain contracts
+- System to support the coordination of cross-chain contract and applications
+  - Explorer to track the cross-chain status
+
+
+# Initial Committers
+- https://github.com/bluele
+- https://github.com/3100
+- https://github.com/siburu
+
+
+# Pre-existing repository
+- https://github.com/datachainlab/fabric-ibc
+- https://github.com/datachainlab/ibc-solidity
+- https://github.com/datachainlab/corda-ibc
```

Comment:
If we are to bring these repos to the Hyperledger Labs organization, what would each of them be named? Normally, we would name the labs repository after the name of the lab. And if there are multiple repos, they would all start with the same prefix.

Reply:
Thanks Tracy,
If I understood correctly, we need to change their names from original ones.
And it should include labs name (here yui) so that anyone can see these sub-products belong to the same lab, right?

Reply:
We are considering following two options. (1) is what we prefer. However, if it is not possible, we change their name as (2).
(1) Stick to original names.
(2) Put prefix of "yui-" to each name.

Reply:
Why not having a single repo (yui) and put the existing three repos into subdirectories?

Reply:
Thanks Arnaud,
Now I understand that typically each Hyperledger Lab consist one integrated repository.

However, we would like to seek out the possibility of having multiple repositories.
This is because as described in proposal, our IBC modules are configured for each blockchain (Fabric and Besu, and also Corda). Thus each module can be used independently.
For managing commits, I would say it would be much better if we can separate these module when it's possible (like repositories of "Perun").

Let us know the limitations to seek out how we can build code base in the easiest way.

Reply:
@tkuhrt @lehors 
As for repository, we would like to separate three modules as they are self-contained and used independently.
We can put prefix for clarify that they belong to the same lab. Please let us know if it is possible.

Also, I think other two suggestions to be resolved. It would be great if you could check them as well.